### PR TITLE
Corrrupted close sector test

### PR DESCRIPTION
**Pull request by kapi-no**
test to cover yet another corner case

### CODE REVIEW

**Comment by kapi-no** [comment]
cc: @nvlsianpu

File: tests/subsys/fs/nvs/src/main.c [hunk 1]
```diff
@@ -244,6 +244,109 @@ void test_nvs_gc(void)
...
+
+		memset(buf, id_data, sizeof(buf));
+
+		if (i == max_writes - 1) {
+			/* Reset stats. */
+			*flash_write_stat = 0;
+			*flash_erase_stat = 0;
+
+			/* Block write calls and simulate power down
+			 * during sector closing operation.
```
Comment (nvlsianpu) [inline tests/subsys/fs/nvs/src/main.c]: so only a part of a NVS closing ate will be written.

File: tests/subsys/fs/nvs/src/main.c [hunk 2]
```diff
@@ -244,6 +244,109 @@ void test_nvs_gc(void)
...
+			     "nvs_read unexpected failure: %d", len);
+
+		for (u16_t i = 0; i < sizeof(rd_buf); i++) {
+			rd_buf[i] = rd_buf[i] % max_id;
+			buf[i] = id;
+		}
+		zassert_mem_equal(buf, rd_buf, sizeof(rd_buf),
+				"RD buff should be equal to the WR buff");
+
+	}
```
Comment (nvlsianpu) [inline tests/subsys/fs/nvs/src/main.c]: can you add one more write and read operation after that to ensure NVS is capable to do so?

**Comment by kapi-no** [comment]
@nvlsianpu, I addressed your comments and rebased the PR to deal with merge conflicts.

**Comment by kapi-no** [comment]
CI has failed because of some kernel tests. It seems unrelated to this PR, how should I fix it?

**Comment by nvlsianpu** [comment]
recheck

**Review by nvlsianpu — APPROVED** [review]

**Comment by kapi-no** [comment]
@carlescufi, @nashif could you merge this PR?

**Review by carlescufi — APPROVED** [review]
